htdocs/include/findusers.php

```diff
@@ -1,526 +1,526 @@
+		$order_select = new icms_form_elements_Select(_MA_USER_ORDER, "user_order", @$_POST["user_order"]);
+		$order_select->addOptionArray(array("ASC"=>_MA_USER_ASC,"DESC"=>_MA_USER_DESC));
+
+		$form->addElement($sort_select);
+		$form->addElement($order_select);
+	}
+
+	$form->addElement( new icms_form_elements_Text(_MA_USER_LIMIT, "limit", 6, 6, empty($_REQUEST["limit"]) ? 50 : (int) ($_REQUEST["limit"])) );
+	$form->addElement( new icms_form_elements_Hidden("mode", $mode) );
+	$form->addElement( new icms_form_elements_Hidden("target", @$_REQUEST["target"]) );
+	$form->addElement( new icms_form_elements_Hidden("multiple", @$_REQUEST["multiple"]) );
+	$form->addElement( new icms_form_elements_Hidden("token", $token) );
+	$form->addElement( new icms_form_elements_Button("", "user_submit", _SUBMIT, "submit") );
+
+	$acttotal = $user_handler->getUserCountByGroupLink(array(), new icms_db_criteria_Item('level', 0, '>'));
+	$inacttotal = $user_handler->getUserCountByGroupLink(array(), new icms_db_criteria_Item('level', 0, '<='));
+	echo "</html><body>";
+	echo "<h2 style='text-align:"._GLOBAL_LEFT.";'>"._MA_USER_FINDUS." - ".$modes[$mode]."</h2>";
+	$modes_switch = array();
+
+	foreach ($modes as $_mode => $title) {
+		if ($mode == $_mode) continue;
+		$modes_switch[] = "<a href='findusers.php?target=".htmlspecialchars(@$_REQUEST["target"], ENT_QUOTES)."&amp;multiple=".htmlspecialchars(@$_REQUEST["multiple"], ENT_QUOTES)."&amp;token=".htmlspecialchars($token, ENT_QUOTES)."&amp;mode={$_mode}'>{$title}</a>";
+	}
+
+	echo "<h4>".implode(" | ", $modes_switch)."</h4>";
+	echo "(".sprintf(_MA_USER_ACTUS, "<span style='color:#ff0000;'>$acttotal</span>")." ".sprintf(_MA_USER_INACTUS, "<span style='color:#ff0000;'>$inacttotal</span>").")";
+	$form->display();
+
+} else {
+	$limit = empty($_POST['limit']) ? 50 : (int) ( $_POST['limit'] );
+	$start = (int) ( @$_POST['start'] );
+
+	if (!isset($_POST["query"])) {
+		$criteria = new icms_db_criteria_Compo();
+		foreach (array_keys($items_match) as $var) {
+			if (!empty($_POST[$var])) {
+				$match = (!empty($_POST["{$var}_match"])) ? (int) ($_POST["{$var}_match"]) : XOOPS_MATCH_START;
+				$value = str_replace("_", "\\\_", icms_core_DataFilter::addSlashes(trim($_POST[$var])));
+				switch ($match) {
+					case XOOPS_MATCH_START:
+						$criteria->add(new icms_db_criteria_Item($var, $value.'%', 'LIKE'));
+						break;
+					case XOOPS_MATCH_END:
+						$criteria->add(new icms_db_criteria_Item($var, '%'.$value, 'LIKE'));
+						break;
+					case XOOPS_MATCH_EQUAL:
+						$criteria->add(new icms_db_criteria_Item($var, $value));
+						break;
+					case XOOPS_MATCH_CONTAIN:
+						$criteria->add(new icms_db_criteria_Item($var, '%'.$value.'%', 'LIKE'));
+						break;
+				}
+			}
+		}
+
+		if (!empty($_POST['url'])) {
+			$url = formatURL(trim(filter_input(INPUT_POST, $_POST['url'], FILTER_SANITIZE_URL)));
+			$criteria->add(new icms_db_criteria_Item('url', $url.'%', 'LIKE'));
```

Comment:
If url contains `%` character, criteria will return incorrect results. So, every `%` in url needs `\`. Source -> https://www.techonthenet.com/mysql/like.php

Reply:
ok, so that does mean that I need to transform every '%' to '\%' ?

Reply:
@MekDrop hehe, github removed my escape character.I need to transform every percentage to baskslash-percentage by doing a replace?

Reply:
@MekDrop do you think it is best to define a new icms_core_Datafilter::addSlashesDB function, that escapes for the %, or do I extend the existing icms_core_Datafilter::addSlashes with the extra parameter that so that it mimics the addslashes PHP function?

Reply:
I think probably adding extra parameter would be easier. But overall I don't see any big differences in these ways - we just need to make it work.